### ISSUE TITLE
Cluster Auto Scaling

### ISSUE THREAD

**Issue by gabegorelick** [awslabs/ecs-refarch-cloudformation#107]
ECS Cluster Auto Scaling [is now GA](https://aws.amazon.com/blogs/aws/aws-ecs-cluster-auto-scaling-is-now-generally-available/). It would make sense to add that once support is [added in CloudFormation](https://github.com/aws/containers-roadmap/issues/76#issuecomment-561435496).  

Cluster Auto Scaling may also remove the need for a dedicated [container instance draining lifecycle hook](https://github.com/aws-samples/ecs-refarch-cloudformation/blob/a257e226b33bd9d2a721e5afd9d7e8b66dbacfdc/infrastructure/lifecyclehook.yaml).

**Comment by bs-thomas** [comment]
Yes, I was about to ask about this, as capacity providers are now generally available.

I was wondering if there are any planned updates for modifying this sample stack to use capacity providers instead.

CloudFormation seems to support them now:
* https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-capacityprovider-autoscalinggroupprovider.html
* https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-capacityprovider-managedscaling.html

It would be really cool if we can have an example!

Cuz I'm still experiencing issues with unbalanced tasks on different instances:

![image](https://user-images.githubusercontent.com/2464920/90820180-8cb28a80-e363-11ea-88e6-f8b1ea39b653.png)


Thank you very much in advance!
